chia/rpc/wallet_rpc_api.py

```diff
@@ -1089,6 +1101,21 @@ async def did_update_recovery_ids(self, request):
         success = spend_bundle is not None and update_success
         return {"success": success}
 
+    async def did_update_metadata(self, request):
+        wallet_id = int(request["wallet_id"])
+        wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        metadata: Dict[str, str] = {}
+        if "metadata" in request:
+            if type(request["metadata"]) is dict:
+                metadata = request["metadata"]
+        async with self.service.wallet_state_manager.lock:
+            update_success = await wallet.update_metadata(metadata)
+            # Update coin with new ID info
+            spend_bundle = await wallet.create_update_spend()
```

Comment:
```suggestion
            spend_bundle = await wallet.create_update_spend() if update_success else None
```

Reply:
Avoids unnecessarily calling create_update_spend() if the call to `update_metadata` failed.